compiler/objectfmt/FunctionCallData.hpp

```diff
@@ -0,0 +1,52 @@
+/*******************************************************************************
+ * Copyright (c) 2020, 2020 IBM Corp. and others
+ *
+ * This program and the accompanying materials are made available under
+ * the terms of the Eclipse Public License 2.0 which accompanies this
+ * distribution and is available at http://eclipse.org/legal/epl-2.0
+ * or the Apache License, Version 2.0 which accompanies this distribution
+ * and is available at https://www.apache.org/licenses/LICENSE-2.0.
+ *
+ * This Source Code may also be made available under the following Secondary
+ * Licenses when the conditions for such availability set forth in the
+ * Eclipse Public License, v. 2.0 are satisfied: GNU General Public License,
+ * version 2 with the GNU Classpath Exception [1] and GNU General Public
+ * License, version 2 with the OpenJDK Assembly Exception [2].
+ *
+ * [1] https://www.gnu.org/software/classpath/license.html
+ * [2] http://openjdk.java.net/legal/assembly-exception.html
+ *
+ * SPDX-License-Identifier: EPL-2.0 OR Apache-2.0 OR GPL-2.0 WITH Classpath-exception-2.0 OR LicenseRef-GPL-2.0 WITH Assembly-exception
+ *******************************************************************************/
+
+#ifndef TR_FUNCTIONCALLDATA_INCL
+#define TR_FUNCTIONCALLDATA_INCL
+
+#include "objectfmt/OMRFunctionCallData.hpp"
+#include "infra/Annotations.hpp"
+
+namespace TR
+{
+
+class OMR_EXTENSIBLE FunctionCallData : public OMR::FunctionCallDataConnector
+   {
+public:
+
+   FunctionCallData(
```

Comment:
Is the FunctionCallData needed to be short lived object only? Do we need `TR_ALLOC` for the FunctionCallData in case we want to cache the information? One example, I was thinking and currently trying out is in Snippet API where as I mentioned in https://github.com/eclipse/omr/pull/5575/files#r517395961, We can add `print` API and for that if we have cached the function call data to snippet, it will be easier.

Reply:
> Do we need TR_ALLOC for the FunctionCallData in case we want to cache the information?

All `TR_ALLOC` does is add a bunch of `placement new` definitions. You don't technically need it since you _could_ just do something like
```
void *storage = allocateHeapMemory(sizeof(FunctionCallData));
FunctionCallData *fcData = new (storage) FunctionCallData(...);
```
Obviously this is terrible code, but the point being that `TR_ALLOC` doesn't give you any new caching ability, just that it allows cleaner code, ie
```
FunctionCallData *fcData = new (comp->trHeapMemory()) FunctionCallData(...);
```

Reply:
Okk yes, We do not need `TR_ALLOC` for allocating object on the heap. So let me rephrase my question, can we add it to list of object types in TR_Memory so that it can be used in more cleaner code as mentioned by @dsouzai  in previous comment?